osu.Game/Overlays/OptionsOverlay.cs

```diff
@@ -101,17 +105,39 @@ public OptionsOverlay()
                 sidebar = new OptionsSidebar
                 {
                     Width = sidebar_width,
-                    Children = sections.Select(section =>
+                    Children = sidebarButtons = sections.Select(section =>
                         new OptionsSidebar.SidebarButton
                         {
-                            Icon = section.Icon,
-                            Action = () => scrollContainer.ScrollIntoView(section)
+                            Selected = sections[0] == section,
+                            Section = section,
+                            Action = () => scrollContainer.ScrollTo(
+                                scrollContainer.GetChildY(section) - scrollContainer.DrawSize.Y / 2),
                         }
-                    )
+                    ).ToArray()
                 }
             };
         }
 
+        private void ScrollContainer_Scrolled(float value)

```

Comment:
I dislike the use of events when they're not necessary, since they make the order of updates hard to predict. If we want the highlighted section to be based off the current scroll position and not the target position, then we won't get around updating it every frame anyways; so could you just put this logic into `OptionsOverlay.Update` and remove the event?